### PR TITLE
Migrated Radio Button to typescript

### LINKED ISSUE #904: RadioButton Typescript Migration
TypeScript here we come!
We are migrating to typescript, we are doing so component by component

We would love some help in converting some of our component to typescript - we've created a README
https://github.com/mondaycom/monday-ui-react-core/blob/master/TYPESCRIPT_MIGRATION.md

in this PR we expect you to convert the following files

- [ ] RadioButton.jsx
- [ ] RadioButtonConstants.js
- [ ] radioButton-snapshot-tests.jest.js
- [ ] radioButton-tests.jest.js

Good Luck!

### PR DESCRIPTION
**Pull request by Suman94310**
#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [X] PR has description.
- [X] New component is functional and uses Hooks. 
- [X] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [X] Stories include all flows of using the component.
#### Tests
- [X] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

## Description
This P.R. migrates "RadioButton" and it's children component( "Clickable" ) to typescript.
this resolves #904

### CODE REVIEW

File: src/components/Clickable/Clickable.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+const CSS_BASE_CLASS = "monday-style-clickable";
+const bemHelper = BEMClass(CSS_BASE_CLASS);
+
+export interface ClickableProps extends VibeComponentProps {
+  elementType?: keyof JSX.IntrinsicElements | string;
+  className?: string;
+  id?: string;
+  children?: React.ReactNode;
+  role?: string;
```
Comment (orrgottlieb) [inline src/components/Clickable/Clickable.tsx:21]: Please change this to `AriaRole` 🙏

File: src/components/Clickable/Clickable.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  onClick?: (event: KeyboardEvent) => void;
```
Comment (orrgottlieb) [inline src/components/Clickable/Clickable.tsx:22]: This could also be a `MouseEvent`

File: src/components/Clickable/Clickable.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  enableTextSelection?: boolean;
+  onMouseDown?: (event: React.MouseEvent) => void;
+  ariaLabel?: string;
+  ariaHidden?: boolean;
+  ariaHasPopup?: boolean | string;
+  ariaExpanded?: boolean;
+  tabIndex?: string;
+  disabled?: boolean;
+  style?: object;
```
Comment (orrgottlieb) [inline src/components/Clickable/Clickable.tsx:31]: we have CSSProperites type

**Review by orrgottlieb — CHANGES_REQUESTED** [review]
Hi @Suman94310 
thank you so much, i have some small comments 🙏 
great PR!!

**Comment by Suman94310** [comment]
Thanks for reviewing my PR @orrgottlieb, I have updated the PR .

File: src/components/Clickable/Clickable.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import { ELEMENT_TYPES, getTestId } from "../../utils/test-utils";
+import "./Clickable.scss";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+
+const CSS_BASE_CLASS = "monday-style-clickable";
+const bemHelper = BEMClass(CSS_BASE_CLASS);
+
+export interface ClickableProps extends VibeComponentProps {
+  elementType?: keyof JSX.IntrinsicElements | string;
```
Comment (orrgottlieb) [inline src/components/Clickable/Clickable.tsx:17]: nice!

**Review by orrgottlieb — APPROVED** [review]
Thank you @Suman94310 for this awesome PR!
Approved, Will merged and will add the appropriate tag for the issue

**Pull request merged**